minerva-university/CourseSwaps
# Edit/delete my swaps

## Description

Khaled: created endpoints for editing swaps, canceling swaps, and fetching user's swaps
Fady: edited the fetch user's swaps endpoint, created "My swaps" list and all the components within it (course item, buttons, edit form), connected the backend to frontend, and displayed error messages as snack bars. 

## Type of Change

## How Has This Been Tested?
Khaled: done backend testing
Fady: checked functionality of the front end visually and practically (will add testing files)

## How to Run Tests
Run backend tests from the test folder


Frontend: check components and buttons render nicely, error messages are displayed when needed, component is refreshing when appropriate. (will add testing later)

## Checklist:

## Screenshots (if applicable):
<img width="1440" alt="Screenshot 2023-12-13 at 04 23 33" src="https://github.com/minerva-university/CourseSwaps/assets/131557935/910c800e-45f4-4848-a446-a2beaf2c587f">
<img width="456" alt="Screenshot 2023-12-13 at 04 23 54" src="https://github.com/minerva-university/CourseSwaps/assets/131557935/3cefccbf-85df-489a-8dad-971bb015eca4">
<img width="487" alt="Screenshot 2023-12-13 at 04 24 05" src="https://github.com/minerva-university/CourseSwaps/assets/131557935/6e8fc321-a223-4aef-b8f3-8667f7341ccf">
<img width="1397" alt="Screenshot 2023-12-13 at 04 24 32" src="https://github.com/minerva-university/CourseSwaps/assets/131557935/79954870-99a5-46b1-8de8-ef2204d8674d">
<img width="391" alt="Screenshot 2023-12-13 at 04 24 46" src="https://github.com/minerva-university/CourseSwaps/assets/131557935/66a5fe0f-7f5c-4b72-9bab-3f4b787d2bcd">
